### PR TITLE
Reorganize docs

### PR DESCRIPTION
**Pull request by wooorm**
This commit reorganizes and rewrites the advanced docs and also adds
a more beginner friendly introduction to MDX in Getting Started.

The need for these changes comes from the fact that MDX is now about
a year old and has grown a lot since.  In that growing, a lot of the
docs changed and were shuffled around.  And lots of new stuff was
added.  Additionally, with the 1.0.0 release around the corner, MDX
is going stable and is ready for use by people that aren’t familiar
with stuff like ASTs and jargon-y terms.

This branch supersedes the `more-docs` branch.

### CODE REVIEW

**Comment by wooorm** [comment]
It’s probably easiest to review the changes on the [now deployment](https://mdx-wqcwatnedd.now.sh).

**Comment by wooorm** [comment]
There’s also some TODOs in there that I’m not sure what to do with btw!

File: docs/advanced/api.md [hunk 1]
```diff
@@ -66,12 +85,46 @@ export default function MDXContent(props) {
...
+```js
+const fs = require('fs')
+const mdx = require('@mdx-js/mdx')
+
+const mdxText = fs.readFileSync('hello.mdx', 'utf8')
+
+const jsx = mdx.sync(mdxText)
+```
+
+<!-- TODO: links should be updated. Can we also inline this example? -->
```
Comment (johno) [inline docs/advanced/api.md]: Yeah inlining here would be good 👍

File: docs/advanced/plugins.md [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
 
-Since MDX uses the [remark][]/[rehype][] ecosystems, you can use plugins to
-modify the AST at different stages of the transpilation process.  This allows
-you to do powerful things like ensure title case capitalization of your headings
-or automate the creation of a table of contents.
+MDX uses [remark][] and [rehype][] internally.
+These ecosystems focus on plugins that transform syntax trees.
+This may sound very much like what MDX does.
+In fact, that’s what we’re doing: the core of MDX is mostly plugins itself!
+We first use `remark-mdx` to add MDX syntax support and then uses a rehype
```
Comment (johno) [inline docs/advanced/plugins.md]:
```suggestion
We first use `remark-mdx` to add MDX syntax support and then use a rehype
```

File: docs/advanced/plugins.md [hunk 3]
```diff
@@ -68,25 +81,93 @@ const emoji = require('remark-emoji')
...
 })
 ```
 
-The following example ensures that `padSpaceAfter` is only passed as options to
+The above example ensures that `padSpaceAfter` is only passed as options to
 the `emoji` plugin.
 
+## Using retext plugins
+
+[retext][] plugins are like remark and rehype plugins, really useful.
```
Comment (johno) [inline docs/advanced/plugins.md]:
```suggestion
[retext][] plugins are, like remark and rehype plugins, really useful.
```

**Comment by johno** [comment]
This is amazing! I'm going to go ahead and merge this since it's a drastic improvement from what's currently there. I'll write up an issue to address the docs TODOs as we continue working towards v1.

**Pull request merged**